architecture/data-model/class-data-model.puml

```diff
@@ -1,19 +1,27 @@
 @startuml
 'https://plantuml.com/class-diagram
 
+hide circle
+
 package ts {
     class Timesheet {
-        entry : Entry
         project : Project [*]
     }
 
+    class Session {
+        start : xsd:dateTime
+        timesheet : Timesheet
+    }
+
+    Session "1" *-- "*" Entry
+
     class Entry {
+        session : Session
         activity : xsd:string
         hourlyRate : hourlyRate
-        contributor : vf:Provider
-        startTime : xsd:dateTime
-        duration: om2:Duration
-        timesheet : Timesheet
+        vf:provider : foaf:Agent
```

Comment:
`vf:Provider` does not exist as a class. So I have used the `vf:property` property to link direct to a `foaf:Agent`